### PR TITLE
Fixing 'Failing to load dynamic library' on Windows

### PR DESCRIPTION
**Pull request by domesticmouse**
Initial attempt at fixing https://github.com/dart-lang/samples/issues/32

However, I am still hitting the following issue, which I assume is visibility related:

```bash
PS C:\src\dart-samples\ffi\hello_world> dart .\hello.dart
Unhandled exception:
Invalid argument(s): Failed to lookup symbol (127)
#0      DynamicLibrary.lookup (dart:ffi-patch/ffi_dynamic_library_patch.dart:33:29)
#1      main (file:///C:/src/dart-samples/ffi/hello_world/hello.dart:21:8)
#2      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:307:19)
#3      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
```

### CODE REVIEW

**Comment by dcharkes** [comment]
The old doc explains how to deal with the visibility on Windows: https://github.com/dart-lang/samples/tree/3933ad3a77ecb4e70a438e4493b8f63a241ded58/ffi. I don't know how that would translate to CMake.

File: ffi/hello_world/hello.dart [hunk 1]
```diff
@@ -13,8 +13,8 @@ typedef HelloWorld = void Function();
 main() {
   // Open the dynamic library
   var path = "./hello_library/libhello.so";
-  if (Platform.isMacOS) path = "./hello_library/libhello.dylib";
-  if (Platform.isWindows) path = "hello_library\libhello.dll";
+  if (Platform.isMacOS) path = './hello_library/libhello.dylib';
+  if (Platform.isWindows) path = r'hello_library\Debug\hello.dll';
```
Comment (johnpryan) [inline ffi/hello_world/hello.dart]: do the other 2 samples need this change too?

**Comment by domesticmouse** [comment]
I’ll work through the other samples once I get this one working...

File: ffi/hello_world/hello.dart (hunk 1 above)
Comment (domesticmouse) [reply ffi/hello_world/hello.dart]: Yup, I'll do the two other samples in follow up PRs.

**Comment by domesticmouse** [comment]
This now works on Windows, and confirmed to still work on Ubuntu.

**Pull request merged**